src/pyseekdb/client/client_base.py

```
@@ -2546,14 +2546,13 @@ def _collection_hybrid_search(
         # Execute SET statement
         self._execute_query_with_cursor(conn, set_sql, [], use_context_manager)
 
-        # Get SQL query from DBMS_HYBRID_SEARCH.GET_SQL
-        get_sql_query = f"SELECT DBMS_HYBRID_SEARCH.GET_SQL('{table_name}', @search_parm) as query_sql FROM dual"
-        logger.debug(f"Getting SQL query: {get_sql_query}")
+        # Call DBMS_HYBRID_SEARCH.SEARCH directly instead of GET_SQL + SQL execution
+        search_query = f"SELECT DBMS_HYBRID_SEARCH.SEARCH('{table_name}', @search_parm) as search_result FROM dual"
```

Ruff (0.14.14)

Possible SQL injection vector through string-based query construction

(S608)